File: hexagonal_stiffness/find_unique_test.py
# -*- coding: utf-8 -*-
"""
Created on Fri Nov 14 11:28:41 2014

@author: nhpnp3
"""

import numpy as np

CsM = np.round(10*np.random.rand(15))
#CsM = np.array([1,5,5,5,8,5])

print CsM

phi1 = np.linspace(0,10,11)


orilist_old = range(CsM.size)
orilist_new = range(CsM.size)

unique_list1 = []
unique_list2 = []

for cc in xrange(CsM.size):
    
    if np.any(np.array(orilist_old) == cc) == False:
        continue

    for dd in orilist_old:

        if CsM[cc] == CsM[dd]:           
            
            orilist_new.remove(dd)
            unique_list1.append(dd)
            unique_list2.append(cc)
    
    orilist_old = list(orilist_new)

    
print unique_list1
print unique_list2

File: hexagonal_stiffness/find_unique_test.m
% this script generates a list of random integers
% unique_list1 contains indices of repeating elements
% unique_list2 contains group id's for the indices in unique_list1

gsh = round(10*rand(1,15))

orilist_old = 1:length(gsh);
orilist_new = 1:length(gsh);

unique_list1 = [];
unique_list2 = [];

for cc = 1:length(gsh)
    
    if sum(orilist_old == cc) == 0 
        continue
    end
        
    for dd = orilist_old
        
        if all(abs(gsh(:,cc)-gsh(:,dd)) < 1E-10) == 1        
            orilist_new(orilist_new == dd) = [];
            unique_list1 = [unique_list1, dd];
            unique_list2 = [unique_list2, cc];
        end
    end
    
    orilist_old = orilist_new;
    
end

unique_list1
unique_list2